addon/utils/locale.js

```diff
@@ -13,6 +15,10 @@ export default class Locale {
   //  3. Default `rtl` to `false`
   //  4. Ensure `pluralForm` is defined
   constructor(id, container) {
+    if (initializedLocales[id]) {
+      return initializedLocales[id];
+    }
```

Comment:
Can you try removing the `if` around `locale.rebuild()` in `addTranslations`?

I think we need to do some substantial re-architecting anyway to support #255. Probably best to think about the two problems together.